pkg_resources/__init__.py

```diff
@@ -1636,7 +1636,7 @@ def _validate_resource_path(path):
         )
 
     def _get(self, path) -> bytes:
-        if hasattr(self.loader, 'get_data'):
+        if hasattr(self.loader, 'get_data') and self.loader:
```

Comment:
```suggestion
        if self.loader and hasattr(self.loader, 'get_data'):
```
Might make sense to reverse the short-circuit?

Reply:
Never mind, just saw https://github.com/pypa/setuptools/pull/4262#discussion_r1518238830